versioned_docs/version-1.0/cadence-migration-guide/usdc-migration.mdx

```diff
@@ -0,0 +1,336 @@
+to make this transition as technically, operationally, and financially easy as possible.
+Please reach out to a Flow team member in Discord or wherever you can communicate with
+Flow to ask for assistance with this upgrade.
+
+## Direct Import of FiatToken
+
+Some smart contracts import `FiatToken` directly with `import FiatToken from 0xb19436aae4d94622`.
+With [the FLIP](https://github.com/onflow/flips/pull/283)
+to be able to import and use broken types and values, a simple import
+of the broken `FiatToken` contract will not cause a contract to fail type checking,
+so if all your contract does with `FiatToken` is import it to use
+types from it as function arguments or return values, you can simply update
+those types to the corresponding `USDCFlow` types.
+
+This also applies to arguments and return values that are Capabilities
+that point to `FiatToken` vaults. These will also need to be updated
+to the new types.
+
+If your users are expecting to use the old `FiatToken` values with your contract,
+it is important for you to notify them that in order to keep using your contract,
+they need to convert their old `FiatToken` to `USDCFlow`
+via the public `USDCFlow.wrapFiatToken()` function before interacting anymore.
+It might also be useful to provide helpful error messages in functions and transactions
+that direct users to how they can swap their `FiatToken.Vault` for a `USDCFlow.Vault`.
+
+Example:
+```cadence
+// Old Code
+import FiatToken from 0xb19436aae4d94622
+
+pub fun sendFiatTokenToAddress(to: Address, vault: @FiatToken.Vault) {
+    // Get the recipient's public account object
+    let recipient = getAccount(to)
+
+    // Get a reference to the recipient's Receiver
+    let receiverRef = recipient.getCapability(FiatToken.VaultReceiverPubPath)
+        .borrow<&{FungibleToken.Receiver}>()
+        ?? panic("Could not borrow receiver reference to the recipient's Vault")
+
+    // Deposit the withdrawn tokens in the recipient's receiver
+    receiverRef.deposit(from: <-vault)
+}
+
+// New Code
+import USDCFlow from 0x{USDCFlowAddress}
+
+pub fun sendUSDCFlowToAddress(to: Address, vault: @USDCFlow.Vault) {
+    // Get the recipient's public account object
+    let recipient = getAccount(to)
+
+    // Get a reference to the recipient's Receiver
+    let receiverRef = recipient.capabilities.borrow<&{FungibleToken.Receiver}>(USDCFlow.ReceiverPublicPath)
+        ?? panic("Could not borrow receiver reference to the recipient's Vault")
+
+    // Deposit the withdrawn tokens in the recipient's receiver
+    receiverRef.deposit(from: <-vault)
+}
+```
+This does not apply to any contract with stored `FiatToken` values such as vaults though!
+If you store any `FiatToken` values, see the next section.
+
+## Stored `@FiatToken.Vault` Field
+
+If your contract stores ANY types from the `FiatToken` contract, 
+this means that you have state that will not work properly after the Crescendo upgrade.
```

Comment:
I get what's being said here, but it could be unclear based on the content above - it's stated above that the FiatToken vault will pass type checking. Suggesting below based on my interpretation, but I could also be wrong 

```suggestion
If your contract stores **ANY** values defined in `FiatToken` contract, 
**you have state that will not work properly after the Crescendo upgrade.**
```

Reply:
This seems fine to me